### DISCUSSION TITLE
Select Option validation

### DISCUSSION THREAD

**Discussion by davidkaban** [colinhacks/zod#2098] [Q&A] [answered]
hi dev. i have problem with select option.
if use z.string ( we select or not still success)
if use z.string and min 1 ( when null show error but when select still error)

please help me.
love you and team

**Comment by JacobWeisenburger** [comment]
Please send a reproducible code example.

**Comment by davidkaban** [comment]
html
          <label for="" style="font-weight:bold">Select Width</label>
```
          <select class="form-control" v-model="selectWidth">
            <option value="">Please select one</option>
            <option value="a"> a </option>
            <option value="b"> b </option>
          </select>
```
          
  vuejs
  
  const schema = z.object({
  width: z.string({
    invalid_type_error: "Width Is Required"
  }).min(1, {
    message: "Width Is Required"
  })
})

vue js function
function searchSize()
{
  validation.value = schema.safeParse({
    height: selectWidth.value
  })
  console.log(validation.value.success)
}

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.object( {
    width: z.string().min( 1 )
//  ^^^^^
} )

function searchSize () {
    validation.value = schema.safeParse( {
        height: selectWidth.value
//      ^^^^^^
    } )
    console.log( validation.value.success )
}
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by davidkaban** [reply]
im sorry, my false.

const schema = z.object( {
    width: z.string().min( 1 )
} )

function searchSize () {
    validation.value = schema.safeParse( {
        width: selectWidth.value
    } )
    console.log( validation.value.success )
}

**result is false.**

**Comment by JacobWeisenburger** [comment]
Everything is working as expected for me. I'm not very familiar with vue, so there might be a problem with that.
```ts
const schema = z.object( {
    width: z.string().min( 1 )
} )

console.log( schema.safeParse( { width: 'a' } ).success ) // true
console.log( schema.safeParse( { width: 'b' } ).success ) // true
console.log( schema.safeParse( { width: '' } ).success ) // false
```

**Reply by davidkaban** [reply]
oke i will try. thanks for you helped
